ui/imports/shared/stores/TransactionFeesBroker.qml

```diff
@@ -0,0 +1,194 @@
+                "tokenKey": subscriber.tokenKey,
+                "accountAddress": subscriber.accountAddress
+            }
+            isReady: !!subscriber.walletsAndAmounts && 
+                    !!subscriber.tokenKey && 
+                    !!subscriber.accountAddress && 
+                    subscriber.walletsAndAmounts.length &&
+                    subscriber.enabled
+
+            topic: isReady ? JSON.stringify(requestArgs) : ""
+            onResponseChanged: subscriber.feesResponse = response
+        }
+
+        component BurnTokenFeeSubscription: Subscription {
+            required property BurnTokenFeesSubscriber subscriber
+            readonly property var requestArgs: {
+                "type": "burn",
+                "tokenKey": subscriber.tokenKey,
+                "amount": subscriber.amount,
+                "accountAddress": subscriber.accountAddress
+            }
+            isReady: !!subscriber.tokenKey && 
+                    !!subscriber.amount && 
+                    !!subscriber.accountAddress &&
+                    subscriber.enabled
+
+            topic: isReady ? JSON.stringify(requestArgs) : ""
+            onResponseChanged: subscriber.feesResponse = response
+        }
+
+        readonly property Component airdropFeeSubscriptionComponent:  AirdropFeeSubscription {}
+        readonly property Component deployFeeSubscriptionComponent:  DeployFeeSubscription {}
+        readonly property Component selfDestructFeeSubscriptionComponent:  SelfDestructFeeSubscription {}
+        readonly property Component burnFeeSubscriptionComponent:  BurnTokenFeeSubscription {}
+
+        readonly property SubscriptionBroker feesBroker: SubscriptionBroker {
+            onRequest: internal.computeFee(topic)
+        }
+
+        property Connections communityTokensStoreConnections: Connections {
+            target: communityTokensStore
+
+            function onDeployFeeUpdated(ethCurrency, fiatCurrency, errorCode, responseId) {
+                d.feesBroker.response(responseId, { "ethCurrency": ethCurrency, "fiatCurrency": fiatCurrency, "errorCode": errorCode })
+            }
+            function onAirdropFeeUpdated(response) {
+                d.feesBroker.response(response.requestId, response)
+            }
+
+            function onSelfDestructFeeUpdated(ethCurrency, fiatCurrency, errorCode, responseId) {
+                d.feesBroker.response(responseId, { "ethCurrency": ethCurrency, "fiatCurrency": fiatCurrency, "errorCode": errorCode })
+            }
+
+            function onBurnFeeUpdated(ethCurrency, fiatCurrency, errorCode, responseId) {
+                d.feesBroker.response(responseId, { "ethCurrency": ethCurrency, "fiatCurrency": fiatCurrency, "errorCode": errorCode })
+            }
+        }
+
+        function computeFee(topic) {
+            let args = JSON.parse(topic)
+            switch (args.type) {
+                case "airdrop":
+                    computeAirdropFee(topic)
```

Comment:
could we pass `args` here instead of `topic` to avoid parsing the same thing twice?